Emby.Dlna/ContentDirectory/ControlHandler.cs

```diff
@@ -908,6 +908,16 @@ private QueryResult<ServerItem> GetMovieFolders(BaseItem item, User user, StubTy
                 }
             };
 
+            // At this point, array contains six items. If Limit is < 6 then we return too many items, so the extra must be removed.
```

Comment:
I don't really get the stub type and why the limit should work this way. We're not really limiting the number of items, but the stub types?

Reply:
I see what you mean. If that's the case, then the error lies with line #914 / 924 in that it should return 1 and not the array length.

Reply:
It looks to be slightly bigger issue than just this one. 

There are a number of routines that return this array structure for when the stub type does not match the expected content type. eg. GetMovieFolders has Latest, Movies, Collections, Favourites, and Genres.

It's found in three places.

Am looking at the specs.

Reply:
Failing on: Query: ObjectId 0 (Root), BrowseFlag = BrowseDirectChildren, RequestedCount = 0

Specification states (2.5.7.2 Arguments):-
```
TotalMatches: If BrowseFlag is set to “BrowseMetadata”, then TotalMatches MUST be set to 1.
Else if BrowseFlag is set to “BrowseDirectChildren”, then TotalMatches MUST be set to the total
number of objects in the object specified for the Browse() action (independent of the starting index
specified by the StartingIndex argument).
```
So if I'm reading that right, it means that the list of options shouldn't all get returned, and it should be trimmed the amount requested.

Reply:
I looked through the docs and our code. Wouldn't this change show an incorrect childCount for the root folder?

Reply:
Only if the RequestedCount is set, which i don't think it is (i'll double check).

This is more of a making the system meet the operational specs than making it work in Jellyfin,

Reply:
Have double checked - and it doesn't affect general browsing.

e.g, VLC sets a limit of 5000 by default.